source/validate_instruction.cpp

```diff
@@ -131,6 +131,26 @@ spv_result_t CapCheck(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+// Checks that the Resuld <id> is within the valid bound.
+spv_result_t LimitCheckIdBound(ValidationState_t& _,
+                               const spv_parsed_instruction_t* inst) {
+  for (int i = 0; i < inst->num_operands; ++i) {
```

Comment:
Sorry I didn't think of this before.  But instruction already has a result_id field parsed out for you.
Either it's 0 (meaning no result Id), or it's the result_id that the binary parser parsed out.
So no need for a loop here at all.  :-)